lib/pengine/complex.c

```diff
@@ -570,6 +570,21 @@ unpack_requires(pe_resource_t *rsc, const char *value, bool is_default)
                  (is_default? " (default)" : ""));
 }
 
+/*!
+ * \internal
+ * \brief Unpack configuration XML for a given resource
+ *
+ * Unpack the XML object containing a resource's configuration into a new
+ * \c pe_resource_t object.
+ *
+ * \param[in]     xml_obj   XML node containing the resource's configuration
+ * \param[out]    rsc       Where to store the unpacked resource information
+ * \param[in]     parent    Resource's parent (\c NULL if the resource is a
+ *                          primitive)
```

Comment:
parent is NULL for top-level resources of any type. E.g. for a top-level group, the group's parent is NULL, and the individual primitive resources inside it have the group as the parent. For a cloned group, the clone's parent is NULL, the group's parent is the clone, and the group members' parent is the group.

Reply:
Oh yeah, duh. Thanks.